### PR TITLE
support resourceQueryExclude option

### PR DESCRIPTION
**Pull request by ben-lau**
This PR contains a:

- [ ] **bugfix**
- [x] new **feature**
- [ ] **code refactor**
- [x] **test update** 
- [ ] **typo fix**
- [ ] **metadata update**

### Motivation / Use-Case
In #164 , the case of mpeg-ts file will lint as typescript file
### Breaking Changes
No, resourceQueryExclude is optional
### Additional Info
No

### CODE REVIEW

**Review by ricardogobbosouza — CHANGES_REQUESTED** [review]

File: commitlint.config.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 module.exports = {
   extends: ['@commitlint/config-conventional'],
   rules: {
-    "header-max-length": [0],
-    "body-max-line-length": [0],
-    "footer-max-line-length": [0],
+    'header-max-length': [0],
+    'body-max-line-length': [0],
+    'footer-max-line-length': [0],
```
[the remark this answers is not in the document]
Comment (ben-lau) [reply commitlint.config.js:6]: ok, but it's changed by prettier

File: src/index.js [hunk 2]
```diff
@@ -51,7 +54,7 @@ class ESLintWebpackPlugin {
     // execute the linter on the build
     if (!this.options.lintDirtyModulesOnly) {
       compiler.hooks.run.tapPromise(this.key, (c) =>
-        this.run(c, options, wanted, exclude)
+        this.run(c, options, wanted, exclude, options.resourceQueryExclude)
```
[the remark this answers is not in the document]
Comment (ben-lau) [reply src/index.js]: I understand, but the input options has been changed during `apply`, its not same as the type of origin option, so I pass the fourth parameter. However, I can fix it by changing the option type at function `run`

**Comment by ricardogobbosouza** [comment]
Please resolve conflicts too
`declarations` move to `types`

File: src/index.js [hunk 3]
```diff
@@ -69,7 +72,7 @@ class ESLintWebpackPlugin {
 
   /**
    * @param {Compiler} compiler
-   * @param {Options} options
+   * @param {Omit<Options, 'resourceQueryExclude'> & {resourceQueryExclude: RegExp[]}} options
```
[the remark this answers is not in the document]
Comment (ben-lau) [reply src/index.js:75]: The `resourceQueryExclude` in type Options is `RegExp` or `RegExp[]`, but in this function resourceQueryExclude type must be `RegExp[]` which changed in `apply`

[the remark this answers is not in the document]
Comment (ben-lau) [reply src/index.js:75]: image

**Comment by ben-lau** [comment]
> Please, resolve conflicts too. `declarations` moved to `types`

Sure, I will fetch the upstream

**Pull request merged**